pushtx/error.go

```diff
@@ -150,3 +159,72 @@ func ParseBroadcastError(msg *wire.MsgReject, peerAddr string) *BroadcastError {
 	reason := fmt.Sprintf("rejected by %v: %v", peerAddr, msg.Reason)
 	return &BroadcastError{Code: code, Reason: reason}
 }
+
+// MapBackendError maps an error received by another backend for example
+// bitcoind to the neutrino BroadcastError.
+//
+// NOTE: This is needed because the Rebroadcaster is now also used with
+// different node backends like for example bitcoind. This is a temporary
+// workaround to prevent reimplementing the broadcaster for different backends.
+func MapBackendError(err error) *BroadcastError {
+
+	// match is a helper method to easily string match on the error
+	// message.
+	match := func(err error, s string) bool {
+		return strings.Contains(strings.ToLower(err.Error()), s)
+	}
+
+	// Determine if this was an RPC error thrown due to the transaction
+	// already confirming.
+	var rpcTxConfirmed bool
+	if rpcErr, ok := err.(*btcjson.RPCError); ok {
+		rpcTxConfirmed = rpcErr.Code == btcjson.ErrRPCTxAlreadyInChain
+	}
+
+	var code BroadcastErrorCode
+
+	// We only map errors we need for the rebroadcaster to work correctly.
+	switch {
+
+	// Returned by bitcoind when a transaction does not meet the fee
+	// requirements to be accepted into mempool. This happens when the
+	// mempool reached its limits and is now purging low fee transactions.
+	// Increasing the mempool size limit will solve this problem.
+	case match(err, "mempool min fee not met"):
+		fallthrough
+
+	// Returned by bitcoind when a transaction does not meet the fee
+	// requirements to be accepted into mempool because the policy of the
+	// mempool has a higher min relay. Default values for bitcoind is
+	// 1000 sat/kvbyte but this is configurable.
+	case match(err, "min relay fee not met"):
+		code = MempoolInsufficientFee
+
+	// This error is returned when sending a transaction that has already
+	// confirmed to a btcd/bitcoind node over RPC.
+	case rpcTxConfirmed:
+		fallthrough
+
+	// This error is returned when broadcasting a transaction that has
+	// already confirmed to a bitcoind node over the P2P network.
+	case match(err, "txn-already-known"):
+		fallthrough
+
+	// This error is returned when broadcasting/sending a transaction to a
+	// btcd node that already has it in their mempool.
+	case match(err, "already have transaction"):
```

Comment:
Should we copy over most of https://github.com/btcsuite/btcwallet/blob/4383930e9c4c07dba5cf7444b0c08c47bd2be30a/wallet/wallet.go#L3655-L3795 @Roasbeef?

I think that this case should belong to `code = Mempool` as the comment describes

Reply:
I agree, I copied the complete error list over feels like the right way from me too.